compute_tools/src/privilege.rs

```diff
@@ -0,0 +1,58 @@
+use std::str::FromStr;
+
+#[derive(Debug)]
+pub enum Privilege {
```

Comment:
If you add:
```suggestion
#[derive(Debug, serde::Deserialize, serde::Serialize)]
#[serde(rename_all = "UPPERCASE")]
pub enum Privilege
```

then you can use `pub privileges: Vec<Privelege>` in the `SetRoleGrantsRequest` and `SetRoleGrantsResponse` structs, as opposed to parsing it manually later on.

Reply:
Cool, so I guess I can also remove the manual `from_str` and `as_str`. Goof stuff

Reply:
Does it then just fail to serialize in case of some string that is not in the enum and return a similar error as we had before?